ckanext/xloader/jobs.py

```diff
@@ -159,57 +162,61 @@ def xloader_data_into_datastore_(input, job_dict):
 
     # fetch the resource data
     logger.info('Fetching from: {0}'.format(url))
+    tmp_file = get_tmp_file(url)
+    length = 0
+    m = hashlib.md5()
     try:
         headers = {}
         if resource.get('url_type') == 'upload':
             # If this is an uploaded file to CKAN, authenticate the request,
             # otherwise we won't get file from private resources
             headers['Authorization'] = api_key
 
-        def get_url():
-            return requests.get(
-                url,
-                headers=headers,
-                timeout=DOWNLOAD_TIMEOUT,
-                verify=SSL_VERIFY,
-                stream=True,  # just gets the headers for now
-                )
-        response = get_url()
-
-        if response.status_code == 202:
-            # Seen: https://data-cdfw.opendata.arcgis.com/datasets
-            # In this case it means it's still processing, so do retries.
-            # 202 can mean other things, but there's no harm in retries.
-            wait = 1
-            while wait < 120 and response.status_code == 202:
-                logger.info('Retrying after {}s'.format(wait))
-                time.sleep(wait)
-                response = get_url()
-                wait *= 3
-        response.raise_for_status()
+        response = get_response(url, headers)
 
         cl = response.headers.get('content-length')
         if cl and int(cl) > MAX_CONTENT_LENGTH:
-            error_msg = 'Resource too large to download: ' \
-                '{cl} > max ({max_cl}).' \
-                .format(cl=cl, max_cl=MAX_CONTENT_LENGTH)
-            logger.warning(error_msg)
-            raise JobError(error_msg)
+            raise DataTooBigError()
 
         # download the file to a tempfile on disk
-        filename = url.split('/')[-1].split('#')[0].split('?')[0]
-        tmp_file = tempfile.NamedTemporaryFile(suffix=filename)
-        length = 0
-        m = hashlib.md5()
         for chunk in response.iter_content(CHUNK_SIZE):
             length += len(chunk)
             if length > MAX_CONTENT_LENGTH:
-                raise JobError(
-                    'Resource too large to process: {cl} > max ({max_cl}).'
-                    .format(cl=length, max_cl=MAX_CONTENT_LENGTH))
+                raise DataTooBigError
             tmp_file.write(chunk)
             m.update(chunk)
+        data['datastore_contains_all_records_of_source_file'] = True
 
+    except DataTooBigError:
+        tmp_file.close()
+        cl = response.headers.get('content-length')
+        message = 'Data too large to load into Datastore: ' \
+                    '{cl} bytes > max {max_cl} bytes.' \
+                    .format(cl=cl or length, max_cl=MAX_CONTENT_LENGTH)
+        logger.warning(message)
+        file_format = resource.get('format')
+        if MAX_EXCERPT_LINES > 0:
```

Comment:
Instead of indenting this whole block, I would add a guard condition:

```
if MAX_EXCERPT_LINES <= 0:
    raise JobError(message)
```

Like that, the rest of the code can be un-indented by 1 level and it makes the code better readable.